lib/_stream_writable.js

```diff
@@ -590,6 +590,13 @@ Writable.prototype.end = function(chunk, encoding, cb) {
   // Ignore unnecessary end() calls.
   if (!state.ending)
     endWritable(this, state, cb);
+  else if (typeof cb === 'function') {
+    if (!state.finished) {
+      this.once('finish', cb);
```

Comment:
My only concern with this is that it may trigger the max listener warning depending on how many time `end()` called. User may have no control over this if it is not their code.

Reply:
I think the common case is one calls `end()` once or maybe twice unnecessarily when it's unclear who ended it. Max listener warning is like 11?

Doing a custom list of callbacks seems a bit overkill for this? Or?

Reply:
I think it would be fine for the warning to appear, I would not worry about it.